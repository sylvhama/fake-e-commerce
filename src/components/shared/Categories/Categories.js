import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const Nav = styled.nav`
  padding: 0 ${props => props.theme.spacing.sm};
  border-top: 1px solid ${props => props.theme.color.lightGrey};
  min-height: ${props => props.theme.size.categories};
  background-color: ${props => props.theme.color.white};
  display: flex;
  justify-content: center;
  align-items: center;
`;

const HorizontalScroll = styled.div`
  display: flex;
  flex-wrap: nowrap;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
`;

const StyledNavLink = styled(NavLink)`
  margin-right: ${props => props.theme.spacing.lg};
  font-size: ${props => props.theme.font.p};
  color: ${props => props.theme.color.lightGrey};
  text-decoration: none;
  flex-shrink: 0;
  &.active {
    color: ${props => props.theme.color.primary};
    font-weight: bold;
  }
  &:last-child {
    margin-right: 0;
  }
`;

const Categories = ({ categories = [], ...rest }) => (
  <Nav {...rest}>
    <HorizontalScroll>
      <StyledNavLink to="/" exact>
        All
      </StyledNavLink>
      {categories.map(category => (
        <StyledNavLink key={category.id} to={`/category/${category.id}`}>
          {category.name}
        </StyledNavLink>
      ))}
    </HorizontalScroll>
  </Nav>
);

Categories.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })
  )
};

export default Categories;
