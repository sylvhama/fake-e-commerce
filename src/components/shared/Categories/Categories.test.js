import React from 'react';
import 'jest-styled-components';

import { renderWithTheme, wrapWithRouter } from '../../../tests/helpers';

import Categories from './Categories';

test('Categories renders correctly', () => {
  const rendered = renderWithTheme(
    wrapWithRouter(<Categories categories={[{ id: '1', name: 'test' }]} />)
  ).toJSON();
  expect(rendered).toMatchSnapshot();
});
