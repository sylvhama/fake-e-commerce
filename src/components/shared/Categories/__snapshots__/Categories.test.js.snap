// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
.c0 {
  padding: 0 8px;
  border-top: 1px solid #ddd;
  min-height: 40px;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.c2 {
  margin-right: 24px;
  font-size: 1rem;
  color: #ddd;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2.active {
  color: #ea352d;
  font-weight: bold;
}

.c2:last-child {
  margin-right: 0;
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <a
      aria-current="page"
      className="c2 active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      All
    </a>
    <a
      aria-current={null}
      className="c2"
      href="/category/1"
      onClick={[Function]}
    >
      test
    </a>
  </div>
</nav>
`;
